src/wl_init.c

```diff
@@ -1154,8 +1155,11 @@ int _glfwPlatformInit(void)
     _glfwInitTimerPOSIX();
 
     _glfw.wl.timerfd = -1;
-    if (_glfw.wl.seatVersion >= 4)
+    if (_glfw.wl.seatVersion >= 4) {
         _glfw.wl.timerfd = timerfd_create(CLOCK_MONOTONIC, TFD_CLOEXEC);
+	int flags = fcntl(_glfw.wl.timerfd, F_GETFL);
+	fcntl(_glfw.wl.timerfd, F_SETFL, flags | O_NONBLOCK);
+    }
```

Comment:
Instead of that you can pass the `TFD_NONBLOCK` flag to `timerfd_create()`:
```suggestion
    if (_glfw.wl.seatVersion >= 4)
        _glfw.wl.timerfd = timerfd_create(CLOCK_MONOTONIC, TFD_CLOEXEC | TFD_NONBLOCK);
```